### PR TITLE
New Docker compose environment for developing Webpagetest

### PR DESCRIPTION
**Pull request by sammeboy635**
Created a docker-compose container for building Webpagetest for a development environment. Some key features so far:
- Built with Nginx and PHP-Fpm7.4
- Docker Hot-Reloading (Edits to the files are live and don't require rebuild) 
- Xdebug (PHP Debugging Server)
- Quick Docker Builds
- Both containers combined are under a gigabyte in size

I created a README.me explaining how to setup and debug some common issues.

Please Review and any feedback is welcome!

Here are a few screen grabs of some key details:

webpagtestspace
successful-build
xdebug

running shot

### CODE REVIEW

**Comment by deathbearbrown** [comment]
<gasp> It's working!! 
This is great work @sammeboy635 !
I'm going to do some functional testing and then I'll go over the code.

File: docker/local/README.md [hunk 1]
```diff
@@ -1,7 +1,161 @@
...
+  docker-compose up
+```
+
+Start Any Web Browser and navigate to "localhost" to see the Webpagetest homepage. To check if webpagetest is working correctly please go down to "Webpagetest Installation Check". Another resource down below is a setup guide to "Debugging PHP with XDebug on VScode" with this docker container.
+
+
+
+## Running Agent with Server
+
+Follow the guide on how to install Wptagent (https://github.com/WPO-Foundation/wptagent). Once the Agent is working, which can be confirmed by running a local test with...
```
Comment (deathbearbrown) [inline docker/local/README.md:33]: is this readme missing for the agent?

Comment (sammeboy635) [reply docker/local/README.md:33]: > 

@deathbearbrown Thanks so much!

**Comment by sammeboy635** [comment]
> It's working!! This is great work @sammeboy635 ! I'm going to do some functional testing and then I'll go over the code.
@deathbearbrown Thanks so much!!
@deathbearbrown Thanks so much, I hope the team for Webpagetest will like this new docker image!

**Comment by deathbearbrown** [comment]
reviewing might bleed over to tomorrow, I just got caught up with something <3

**Review by deathbearbrown — APPROVED** [review]
@tkadlec Approving this. I've been using it all week and it's been really helpful to find private instance bugs

**Comment by sammeboy635** [comment]
@deathbearbrown I added the WPTagent to the Webpagetest docker-compose.yml. Two separate containers just required a lot of manual work with the network configurations of the bridged networks to allow them to communicate properly (which is not simple). This way with one docker-compose.yml, all we have is one "docker-compose up" and everything just works as intended.

**Pull request merged**